src/util/showIfAuthenticated.js

```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import { firebaseConnect } from 'react-redux-firebase'
+
+const showIfAuthenticated = DecoratedComponent => {
+  const OnlyShowIfAuthenticated = ({ firebase, ...otherProps }) => {
+    if (firebase.auth().currentUser) {
+      return <DecoratedComponent { ...otherProps } />
+    }
+
+    return <div />
+  }
+
+  return firebaseConnect([
+    'auth'
+  ])(OnlyShowIfAuthenticated)
+}
```

Comment:
saw we would be reusing this logic, so I pulled it into a HOC so we can just decorate whichever components need it. was thinking we may also want a version of this that shows an alternative component when not authenticated.

Reply:
I love it! HOCs are still a little new for me, but I knew there had to be a "react" way of handling this sort of logic! So clean